fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailViewModel.java

```diff
@@ -269,25 +271,19 @@ public void endElection(Election election) {
     ElectionEnd electionEnd =
         new ElectionEnd(election.getId(), laoId, election.computerRegisteredVotes());
 
-    MessageGeneral msg = new MessageGeneral(mKeyManager.getMainKeyPair(), electionEnd, mGson);
-
     Log.d(TAG, PUBLISH_MESSAGE);
     Disposable disposable =
-        mLAORepository
-            .sendPublish(channel, msg)
-            .subscribeOn(Schedulers.io())
-            .observeOn(AndroidSchedulers.mainThread())
-            .timeout(5, TimeUnit.SECONDS)
+        networkManager
+            .getMessageSender()
+            .publish(keyManager.getMainKeyPair(), channel, electionEnd)
             .subscribe(
-                answer -> {
-                  if (answer instanceof Result) {
-                    Log.d(TAG, "ended election successfully");
-                    endElectionEvent();
-                  } else {
-                    Log.d(TAG, "failed to end the election");
-                  }
+                () -> {
+                  Log.d(TAG, "ended election successfully");
+                  endElectionEvent();
```

Comment:
outside the scope of this PR, but I'm not sure if the UI should do much until we receive the broadcast back.

Reply:
Right, but as you said, outside the already large scope of this PR